src/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -75,15 +74,22 @@ public void GetHostByName_HostName_GetHostByAddr()
 
             IPHostEntry hostEntry1 = NameResolutionPal.GetHostByName(hostName);
             Assert.NotNull(hostEntry1);
-            IPHostEntry hostEntry2 = NameResolutionPal.GetHostByAddr(hostEntry1.AddressList[0]);
-            Assert.NotNull(hostEntry2);
 
             IPAddress[] list1 = hostEntry1.AddressList;
-            IPAddress[] list2 = hostEntry2.AddressList;
+            Assert.InRange(list1.Length, 1, Int32.MaxValue);
 
-            for (int i = 0; i < list1.Length; i++)
+            foreach (IPAddress addr1 in list1)
             {
-                Assert.NotEqual(-1, Array.IndexOf(list2, list1[i]));
+                IPHostEntry hostEntry2 = NameResolutionPal.GetHostByAddr(addr1);
```

Comment:
Actually, this can fail as well depending on the host name and name server configuration. 
Simplifying the test would make it more resilient. (E.g. calling GetHostByAddr on one of the common configuration Http servers hosted in Azure maybe or even a generic site such as bing.com.)

A UnitTest would be better suited to test our wrapper code by returning simulated OS API responses.